docker/Dockerfile

```diff
@@ -0,0 +1,105 @@
+FROM centos:centos7.4.1708
+
+ENV GOSU_VERSION 1.10
+ENV GOSU_ARCH amd64
+ENV GOSU_URL https://github.com/tianon/gosu/releases/download
+ENV GOSU_APP ${GOSU_URL}/${GOSU_VERSION}/gosu-${GOSU_ARCH}
+ENV GOSU_ASC ${GOSU_URL}/${GOSU_VERSION}/gosu-${GOSU_ARCH}.asc
+ENV DB4_VERSION 4.8.30.NC
+ENV DB4_HASH 12edc0df75bf9abd7f82f821795bcee50f42cb2e5f76a6a281b85732798364ef
+ENV DB4_URL http://download.oracle.com/berkeley-db/db-${DB4_VERSION}.tar.gz
+ENV REPO_URL https://github.com/bitcoin/bitcoin
+ENV BRANCH 0.16
```

Comment:
Is it worth all these being [`ARG`s](https://docs.docker.com/engine/reference/builder/#arg) to allow parameterization without having to edit the Dockerfile?